### ISSUE TITLE
Must delete current data to view updated driver schema

### ISSUE THREAD

**Issue by rmweir** [rancher/rancher#17712]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Run rancher locally.
2. Go to https://YOURHOST/v3/schemas/amazonElasticContainerServiceConfig.
3. Add a driver flag in eks_driver.go and save.
4. Rerun rancher.
5. Go to https://YOURHOST/v3/schemas/amazonElasticContainerServiceConfig.

**Result:**
The schema will not update to reflect the changes. This would have to a user if they were to update rancher without restarting kubernetes.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
master branch 1/29/19

**Comment by cjellick** [comment]
@erikwilson to validate

**Comment by erikwilson** [comment]
To validate I first verified the bug in v2.1.6, viewed the schema at `/v3/schemas/googleKubernetesEngineConfig`, added a 'hello-world' flag to gke_driver.go, and recompiled & restarted the server. As expected the schema was not updated to include the new flag.

I then tried on master (0bbc6c528ca08287f08ab275947773c813ff3fb3) repeating the same process, after recompiling & restarting the schema now contains the new flag.

**Comment by cjellick** [comment]
Can you take the validation one step futher and verify it works properly for external kontainer engine drivers? They are now pluggable like node drivers. @nathan-jenan-rancher has an example driver that is a dupe GKE but external. Can you run the same test (adding a hello world flag) with that?
The difference is that you should not have to restart rancher. Updating the pluggable driver to point to the newer version of the driver should trigger it.

I think this actually goes beyond the scope of the original bug report, but its still important to test.

**Comment by erikwilson** [comment]
I was unable to trigger a schema update with pluggable drivers without removing and re-adding the driver. 

I compiled and added the driver at https://github.com/nathan-jenan-rancher/kontainer-engine-example-driver, then modified gke_drive.go in vendor and updated the hash. 

The driver re-download but a new schema failed to appear. I restarted the server but the schema still failed to update. 

I removed the driver and re-installed and the new schema appeared.

**Comment by cjellick** [comment]
@erikwilson to revalidate

**Comment by erikwilson** [comment]
I was able to validate that the built-in drivers and pluggable drivers will update the schema on change. 

To validate the built-in driver I modified the vendor gke_driver with a flag and restarted rancher, and the schema updated.

To validate the pluggable driver I installed our example driver at rancher/kontainer-engine-driver-example and verified the schema. After recompiling the driver the checksum was updated, but the driver did not re-download. After changing the driver URL the driver re-download and the schema was updated.
